libs/vkd3d/command.c

```diff
@@ -3559,11 +3559,129 @@ static VkPipelineStageFlags vk_render_pass_barrier_from_view(struct d3d12_comman
 static void d3d12_command_list_track_resource_usage(struct d3d12_command_list *list,
         struct d3d12_resource *resource, bool perform_initial_transition);
 
+static void d3d12_command_list_update_subresource_data(struct d3d12_command_list *list,
+        struct d3d12_resource *resource, VkImageSubresourceLayers subresource)
+{
+    struct vkd3d_subresource_tracking *entry;
+    VkImageAspectFlags aspect, aspect_mask;
+    uint32_t i;
+
+    assert(resource->flags & VKD3D_RESOURCE_LINEAR_STAGING_COPY);
+    aspect_mask = subresource.aspectMask;
+
+    while (aspect_mask)
+    {
+        aspect = aspect_mask & -aspect_mask;
+
+        for (i = 0; i < list->subresource_tracking_count; )
+        {
+            entry = &list->subresource_tracking[i];
+
+            if (entry->resource == resource && entry->subresource.aspectMask == aspect && entry->subresource.mipLevel == subresource.mipLevel)
+            {
+                if (entry->subresource.baseArrayLayer <= subresource.baseArrayLayer &&
+                        entry->subresource.baseArrayLayer + entry->subresource.layerCount >= subresource.baseArrayLayer + subresource.layerCount)
+                {
+                    return;
```

Comment:
Should this really be a return? If there are multiple aspects we might get a full coverage here, but not in other aspects.

Reply:
oversight, this didn't iterate over the aspects at first.